src/hdx_cli_toolkit/cli.py

```diff
@@ -734,3 +744,130 @@ def remove_extras_key(
         output_path = make_path_unique(output_path)
         status = write_dictionary(output_path, output_rows, append=False)
         print(status, flush=True)
+
+
+@hdx_toolkit.command(name="scan")
+@click.option(
+    "--hdx_site",
+    type=click.Choice(["stage", "prod"]),
+    is_flag=False,
+    default="stage",
+    help="an hdx_site value",
+)
+@click.option(
+    "--action",
+    type=click.Choice(["survey", "delete_key", "distribution"]),
+    is_flag=False,
+    default="survey",
+    help="an action to take",
+)
+@click.option("--key", is_flag=False, default="private", help="a key or list of keys")
+@click.option(
+    "--start", is_flag=False, default=0, help="a starting offset for the rows returned by a query"
+)
+@click.option(
+    "--rows",
+    is_flag=False,
+    default=None,
+    help="the number of rows to return, "
+    "the limit for the CKAN API is 1000. If None all rows will be returned",
+)
+@click.option(
+    "--verbose",
+    is_flag=True,
+    default=False,
+    help="if true show diagnostic information in run",
+)
+@click.option(
+    "--output_path",
+    is_flag=False,
+    default=None,
+    help="A file path to export package_search records for datasets",
+)
+@click.option(
+    "--input_path",
+    is_flag=False,
+    default=None,
+    help="A file path to import package_search records for datasets",
+)
+def scan(
+    hdx_site: str = "stage",
+    output_path: Optional[str] = None,
+    input_path: Optional[str] = None,
+    action: str = "survey",
+    start: int = 0,
+    rows: Optional[int] = 0,
+    key: str = "name",
+    verbose: bool = False,
+):
+    """Scan all of HDX and perform an action, currently supported actions are:
+
+    1. survey - count the number of occurrences of a key or list of keys across datasets in HDX
+
+    2. distribution - calculate the histogram of values for a key across datasets in HDX
+
+    3. delete_key - delete occurrences of a key across all datasets in HDX, this is currently
+    configured so that it only accepts "extras" and "resource._csrf_token" as valid keys to delete
+    """
+    print_banner("Scan HDX")
+    t0 = time.time()
+    fetch_all = False
+    if rows is None:
+        start = 0
+        rows = 1000
+        fetch_all = True
+    if fetch_all and input_path is None:
+        print(
+            "No rows value provided so fetching all data in 1000 row chunks. "
+            "This takes ~10 minutes and generates an 865MB file.",

```

Comment:
It was only 4 minutes for me!

Reply:
Get you and your fancy internet ;-) Anyway 4 is the same as 10 in physicist order of magnitude terms